c/common/src/codingstandards/c/Errno.qll

```diff
@@ -10,14 +10,29 @@ abstract class ErrnoSettingFunction extends Function { }
 
 /*
  * An errno-setting function that return out-of-band errors indicators
+ * as listed in the MISRA standard
  */
 
-class OutOfBandErrnoSettingFunction extends ErrnoSettingFunction {
-  OutOfBandErrnoSettingFunction() {
+class OutOfBandErrnoSettingFunctionMisra extends ErrnoSettingFunction {
+  OutOfBandErrnoSettingFunctionMisra() {
     this.hasGlobalName(["ftell", "fgetpos", "fsetpos", "mbrtowc", "wcrtomb", "wcsrtombs"])
   }
 }
 
+/*
+ * An errno-setting function that return out-of-band errors indicators
+ * as listed in the CERT standard
+ */
+
+class OutOfBandErrnoSettingFunctionCert extends Function {
+  OutOfBandErrnoSettingFunctionCert() {
+    this.hasGlobalName([
+        "ftell", "fgetpos", "fsetpos", "mbrtowc", "mbsrtowcs", "signal", "wcrtomb", "wcsrtombs",
+        "mbrtoc16", "mbrtoc32", "c16rtomb", "c32rtomb"
+      ])
+  }
+}
+
```

Comment:
I'm not sure that all of these functions can be lumped together the way the query is currently written. For example, in the case of `fgetpos` it sets an implementation defined value of `errno` on failure. However, signal will return a positive value, so it's OK to use errno without first checking.